src/mlpack/methods/ann/CMakeLists.txt

```diff
@@ -16,6 +16,7 @@ add_subdirectory(layer)
 add_subdirectory(loss_functions)
 add_subdirectory(convolution_rules)
 add_subdirectory(augmented)
+add_subdirectory(dists)
```

Comment:
Shouldn't we put this in `src/mlpack/core/dists/`?

Reply:
Ah, yeah, sorry I missed the discussion.  I could see it either way---I don't think it's a problem to add it to the `src/mlpack/core/dists/` folder as long as it has the same functionality as the other distributions there, but if you prefer to keep it here that's fine also.

Reply:
If it's not a problem, I will discuss it with Sumedh and decide whether we should move it back to the core/dists folder.

Reply:
Sure, that's fine, and I wouldn't have a particularly strong opinion here since I am not 100% involved in this bit.  My intuition is usually that if we can put code in more generic places, it's easier for reuse later and helps us keep the amount of code we have to maintain down.  Maybe that is possible here, maybe it doesn't make sense.

Reply:
@sumedhghaisas What do you think about this?

Reply:
@akhandait @rcurtin. I agree with @rcurtin, even my first choice is to put this in core/dists and we did plan that if I remember correctly. We are using NormalDistribution to define distribution over the inputs. The reason I suggested to put this distribution here (for now) is, a lot of ANN layer sensitive code was added to the distribution. For example, constructor accepting mean and variance together in 1 matrix of shape [batch_size, 2 * latent_dim], ability for the user to add SoftplusFunction over the variance to make sure its strictly positive, dealing with batching, and backward for LogProb. All these characteristic make this very ANN specific and putting it in core/dists would involved huge efforts to find the middle ground between the two.

Reply:
So, I guess for now we will keep it in the ann folder.

Reply:
Ah, I see what you mean.  Thanks for clarifying and sorry for making you type out the justification since you probably already wrote it in IRC earlier. :)